nifi-extension-bundles/nifi-standard-services/nifi-dbcp-service-bundle/nifi-dbcp-service/src/main/java/org/apache/nifi/dbcp/DBCPConnectionPool.java

```diff
@@ -261,9 +265,13 @@ protected Driver getDriver(final String driverName, final String url) {
             return DriverManager.getDriver(url);
         } catch (final SQLException e) {
             // In case the driver is not registered by the implementation, we explicitly try to register it.
+            // deregister existing driver
             try {
-                final Driver driver = (Driver) clazz.getDeclaredConstructor().newInstance();
-                DriverManager.registerDriver(driver);
+                if (registeredDriver != null) {
+                    DriverManager.deregisterDriver(registeredDriver);
+                }
+                registeredDriver = (Driver) clazz.getDeclaredConstructor().newInstance();
```

Comment:
Modern JDBC drivers use Java's Service Loader mechanism to load the driver class automatically. Manual instantiation is rarely required and this code in the catch block is more of a ‘just-in-case’ fallback.

May I ask what JDBC driver are you using that requires manual loading?

I'm afraid, we cannot control the automatically registered drivers. Therefore, even if we add a deregistration mechanism for the manual-loading case, the issue would still persist for the majority of the use cases.